scripts/gen_eval_files.lua

```diff
@@ -13,6 +31,51 @@ local LUA_KEYWORDS = {
   ['repeat'] = true,
 }
 
+local API_TYPES = {
+  Window = 'integer',
+  Tabpage = 'integer',
+  Buffer = 'integer',
+  Boolean = 'boolean',
+  Object = 'any',
+  Integer = 'integer',
+  String = 'string',
+  Array = 'any[]',
+  LuaRef = 'function',
+  Dictionary = 'table<string,any>',
+  Float = 'number',
+  void = '',
+}
+
+--- Convert an API type to Lua
+--- @param t string
+--- @param name string
+--- @return string
+local function api_type(t, name)
+  if t:match('^ArrayOf%(([z-aA-Z]+), %d+%') then
+    print(t:match('^ArrayOf%(([z-aA-Z]+), %d+%'))
+  end
+  local as0 = t:match('^ArrayOf%((.*)%)')
+  if as0 then
+    local as = vim.split(as0, ', ', { plain = true })
+    return api_type(as[1], name)..'[]'
+  end
+
+  local d = t:match('^Dict%((.*)%)')
+  if d then
+    -- TODO(lewis6991): render proper types from keysets.h. This is already done in
+    -- gen_api_dispatch.lua
+    -- return 'vim.api.'..d..'_'..name
+    return 'table<string,any>'
```

Comment:
Are we duplicating logic from gen_api_dispatch? Maybe one can require() the other?

Reply:
Not duplicating any logic AFAIK, this todo is gone now.